journey-map/public/render.mjs

```diff
@@ -126,20 +127,39 @@ const renderTransitions = (journeyMap, formData) => {
 
         const eventsByTarget = {};
         Object.entries(events).forEach(([eventName, def]) => {
-            const target = resolveEventTarget(def, formData);
+            const { targetJourney, targetState } = resolveEventTarget(def, formData);
+            const target = targetJourney ? `${targetJourney}__${targetState}` : targetState;
 
-            if (errorStates.includes(target) && !formData.has('includeErrors')) {
+            if (errorJourneys.includes(targetJourney) && !formData.has('includeErrors')) {
                 return;
             }
-            if (failureStates.includes(target) && !formData.has('includeFailures')) {
+            if (failureJourneys.includes(targetJourney) && !formData.has('includeFailures')) {
                 return;
             }
+            if (targetJourney && otherMaps && otherMaps[targetJourney]) {
+
+            }
 
             if (!states.includes(target)) {
                 states.push(target);
             }
+
             eventsByTarget[target] = eventsByTarget[target] || [];
             eventsByTarget[target].push(eventName);
+
+            if (!journeyMap[target]) {
+                if (targetJourney) {
+                    journeyMap[target] = {
+                        response: {
+                            type: 'journeyTransition',
+                            targetJourney,
+                            targetState,
+                        }
```

Comment:
We add a fictional `journeyTransition` node when we encounter a journey transition